cli/tests/test_branch_command.rs

```diff
@@ -128,6 +128,29 @@ fn test_branch_move() {
     insta::assert_snapshot!(stderr, @"");
 }
 
+#[test]
+fn test_branch_set_create() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    let stderr = test_env.jj_cmd_failure(&repo_path, &["branch", "set", "foo"]);
+    insta::assert_snapshot!(stderr, @r###"
+    Error: No such branch: foo
+    Hint: Use `jj branch create` or `jj branch set --create` to create it.
+    "###);
+
+    let (_stdout, stderr) = test_env.jj_cmd_ok(&repo_path, &["branch", "set", "foo", "--create"]);
+    insta::assert_snapshot!(stderr, @"");
+
+    test_env.jj_cmd_ok(&repo_path, &["new"]);
+    let stderr = test_env.jj_cmd_failure(&repo_path, &["branch", "create", "foo"]);
+    insta::assert_snapshot!(stderr, @r###"
+    Error: Branch already exists: foo
+    Hint: Use `jj branch set` to update it.
+    "###);
```

Comment:
nit: isn't this redundant with the test on lines 108-113?

Reply:
Yes that's correct, I will amend this